test/tools/ossfuzz/protoToYul.cpp

```diff
@@ -876,76 +877,123 @@ void ProtoConverter::visit(Statement const& _x)
 	}
 }
 
-void ProtoConverter::visit(Block const& _x)
+void ProtoConverter::openScope(vector<string>&& _x)
+{
+	m_scopes.push({});
+	for (auto const& i: _x)
+	{
+		m_scopes.top().insert(i);
+		m_variables.push_back(i);
+	}
+}
+
+void ProtoConverter::closeScope()
+{
+	for (auto const& var: m_scopes.top())
+	{
+		unsigned numErased = m_variables.size();
+		m_variables.erase(remove(m_variables.begin(), m_variables.end(), var), m_variables.end());
+		numErased -= m_variables.size();
+		yulAssert(numErased == 1, "Proto fuzzer: More than one variable went out of scope");
+	}
+	m_scopes.pop();
+}
+
+void ProtoConverter::addToScope(vector<string>&& _x)
+{
+	for (string const& i: _x)
+	{
+		m_variables.push_back(i);
+		m_scopes.top().insert(i);
+	}
+}
+
+void ProtoConverter::visit(Block const& _x, vector<string>&& _y)
 {
+	openScope(move(_y));
 	if (_x.statements_size() > 0)
 	{
-		m_numVarsPerScope.push(0);
 		m_output << "{\n";
 		for (auto const& st: _x.statements())
 			visit(st);
 		m_output << "}\n";
-		m_numLiveVars -= m_numVarsPerScope.top();
-		m_numVarsPerScope.pop();
 	}
 	else
 		m_output << "{}\n";
+	closeScope();
 }
 
-void ProtoConverter::visit(SpecialBlock const& _x)
+void ProtoConverter::visit(SpecialBlock const& _x, vector<string>&& _y)
 {
-	m_numVarsPerScope.push(0);
+	openScope(move(_y));
 	m_output << "{\n";
 	visit(_x.var());
 	if (_x.statements_size() > 0)
 		for (auto const& st: _x.statements())
 			visit(st);
-	m_numLiveVars -= m_numVarsPerScope.top();
-	m_numVarsPerScope.pop();
 	m_output << "}\n";
+	closeScope();
+}
+
+vector<string> ProtoConverter::createVars(unsigned _startIdx, unsigned _endIdx)
+{
+	yulAssert(_endIdx > _startIdx, "Proto fuzzer: Variable indices not in range");
+	string varsStr = dev::suffixedVariableNameList("x_", _startIdx, _endIdx);
+	m_output << varsStr;
+	vector<string> varsVec;
+	boost::split(
```

Comment:
Maybe don't use `suffixedVariableNameList` if you need to split again anyway?

Reply:
I need to use the concatenated version as well to create the LHS of a multi var declaration/assignment.

The `m_output << varsStr` is essentially using the concatenated version for the proto to yul conversion.